evm/src/cpu/kernel/interpreter.rs

```diff
@@ -1117,7 +763,8 @@ impl<'a> Interpreter<'a> {
     }
 
     fn run_dup(&mut self, n: u8) -> anyhow::Result<()> {
-        if n == 0 {
+        assert!(n > 0);
+        if n == 1 {
```

Comment:
So this was buggy before?

Reply:
It wasn't the expected behavior, but it wasn't buggy because `stack_peek` was doing the job. I am removing the comparison altogether, as it is redundant with what `stack_peek` is doing actually.